src/gluonts/model/seasonal_naive/_predictor.py

```diff
@@ -70,16 +70,11 @@ def __init__(
             else get_seasonality(freq)
         )
 
-    def predict(self, dataset: Dataset, **kwargs) -> Iterator[SampleForecast]:
-        for data in dataset:
-            start = pd.Timestamp(data["start"], freq=self.freq)
-            target = np.asarray(data["target"], np.float32)
-            yield self._predict_time_series(start_time=start, target=target)
-
-    def _predict_time_series(
-        self, start_time: pd.Timestamp, target: np.ndarray
-    ) -> SampleForecast:
+    def predict_item(self, item: DataEntry) -> Forecast:
+        target = np.asarray(item["target"], np.float32)
         len_ts = len(target)
+        forecast_start_time = item["start"] + len_ts * item["start"].freq
```

Comment:
I think there is a function called forecast_start in `gluonts.util.pandas`, which you can use.

Reply:
it was in `gluonts.support.pandas`, thx